.github/workflows/ci.yml

```diff
@@ -177,30 +177,60 @@ jobs:
 
   java-unit-tests:
     if: needs.detect-changes.outputs.java-code-changes == 'true'
-    needs: [detect-changes]
-    runs-on: gcp-perf-core-16-default
-    timeout-minutes: 30
+    name: "[UT] ${{ matrix.name }}"
+    needs: [ detect-changes ]
+    timeout-minutes: 25
     permissions: {}  # GITHUB_TOKEN unused in this job
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        group: [ general, zeebe, client ]
+        include:
+          - group: general
+            name: "General Unit Tests"
+            # Here we exclude all modules that are separated in extra jobs below
+            # Example: '[-+]<groupId>:artifactId' -> '-' excludes
+            maven-modules: "'-:zeebe-workflow-engine','-:zeebe-broker','-:camunda-process-test-java','-:zeebe-gateway-rest','-:camunda-client-java','-:zeebe-elasticsearch-exporter','-:zeebe-opensearch-exporter','-:camunda-exporter','-:zeebe-atomix-cluster','-:zeebe-cluster-config'"
+            maven-build-threads: 2
+            maven-test-fork-count: 7
+            runs-on: gcp-perf-core-16-default
+          - group: zeebe
+            name: "Zeebe Big Unit Tests"
+            maven-modules: "':zeebe-workflow-engine',':zeebe-broker',':zeebe-atomix-cluster',':zeebe-cluster-config'"
+            maven-build-threads: 2
+            maven-test-fork-count: 7
+            runs-on: gcp-perf-core-16-default
+          - group: client
+            name: "Camunda API Big Unit Tests"
+            maven-modules: "':camunda-process-test-java',':zeebe-gateway-rest',':camunda-client-java',':zeebe-elasticsearch-exporter',':zeebe-opensearch-exporter',':camunda-exporter'"
+            maven-build-threads: 2
+            maven-test-fork-count: 7
+            runs-on: gcp-perf-core-16-default
     steps:
       - uses: actions/checkout@v4
       - uses: ./.github/actions/setup-build
         with:
+          maven-cache-key-modifier: it-${{ matrix.group }}
           vault-address: ${{ secrets.VAULT_ADDR }}
           vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
           vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
       - uses: ./.github/actions/build-zeebe
+        id: build-zeebe
         with:
-          maven-extra-args: -T1C -PskipFrontendBuild
+          maven-extra-args: -T1C -Dquickly
```

Comment:
:i:  Decrease build time by 1 minute via using the quickly build flag,
   as all necessary checks are done in other jobs